src/polyfills/details/details-en.hbs

```diff
@@ -80,7 +80,7 @@
 		</li>
 		<li>
 			<details>
-				<summary>Example 3</summary>
+				<summary class="h3"><h3>Example 3</h3></summary>
```

Comment:
@duboisp To properly achieve something like this:
```html
<summary><h3>Example 3</h3></summary>
```

You'd need to resort to the [:has() pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:has)... which is dead in the water in terms of browser support.

Another way I can think of would be to use a ``::before`` pseudo-element to add a fake disclosure widget to child heading elements and hope the fake one hides the real one. But IMO that probably wouldn't work in practice. If the heading's line height is substantially taller than the summary's line height, the heading's fake disclosure widget probably wouldn't be able to hide the summary's native one.

If you're thinking about disabling the ``<summary>`` element's native disclosure widget and using ``::before`` to insert a fake one into its first child element (regardless of if it's a heading or normal text)... that wouldn't work out either since  summaries don't typically contain child elements.

Plus according to the HTML spec, the [``<summary>`` element's content model](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-summary-element:concept-element-content-model) allows it to contain intermixed phrasing/heading content. So any CSS solution that doesn't resort to ``:has()`` would be opinionated and go against the spec's content model.

**TL;DR**
I don't think there's a better way of doing this than using a ``h#`` class and an equivalent ``<h#>`` element. IMO the other methods would have big downsides. I also don't think it's worthwhile to add extra CSS selectors to enforce using  equivalent classes/heading elements since that'd result in a ton of extra selectors (25 by my count).